### ISSUE TITLE
use _yaml.pyd parse yaml with error

### ISSUE THREAD

**Issue by poursoul** [yaml/pyyaml#293]
error message:
  File "_yaml.pyx", line 707, in _yaml.CParser.get_single_node (ext\_yaml.c:7710)
  File "_yaml.pyx", line 725, in _yaml.CParser._compose_document (ext\_yaml.c:7969)
  File "_yaml.pyx", line 776, in _yaml.CParser._compose_node (ext\_yaml.c:8663)
  File "_yaml.pyx", line 890, in _yaml.CParser._compose_mapping_node (ext\_yaml.c:10214)
  File "_yaml.pyx", line 776, in _yaml.CParser._compose_node (ext\_yaml.c:8663)
  File "_yaml.pyx", line 890, in _yaml.CParser._compose_mapping_node (ext\_yaml.c:10214)
  File "_yaml.pyx", line 776, in _yaml.CParser._compose_node (ext\_yaml.c:8663)
  File "_yaml.pyx", line 890, in _yaml.CParser._compose_mapping_node (ext\_yaml.c:10214)
  File "_yaml.pyx", line 776, in _yaml.CParser._compose_node (ext\_yaml.c:8663)
  File "_yaml.pyx", line 887, in _yaml.CParser._compose_mapping_node (ext\_yaml.c:10178)
  File "_yaml.pyx", line 905, in _yaml.CParser._parse_next_event (ext\_yaml.c:10424)
  ScannerError: while scanning a plain scalar
    in "<byte string>", line 91, column 13
  found unexpected ':'
    in "<byte string>", line 91, column 26
  Error message is while scanning a plain scalar
    in "<byte string>", line 91, column 13
  found unexpected ':'
============
i found reason because of white space after ':' can't given, for example:
  a:
    b: {c:[d,e]}

this code will get success by used yaml.Parser, and will get error by use _yaml.Parser :(

**Comment by Lexy2** [comment]
Yes, it's a bug, also encountered it. Any spaces after colons can cause either `ScannerError` or `ParserError`

```import yaml
data = yaml.load("d:{1:[18,0,0]}", yaml.CLoader)
```
This works. 

```import yaml
data = yaml.load("d: {1:[18,0,0]}", yaml.CLoader)
```
This doesn't. ScannerError.

```import yaml
data = yaml.load("d: {1:[18,0,0]}", yaml.Loader)
```
This works.

```import yaml
data = yaml.load("d:{1: [18,0,0]}", yaml.CLoader)
```
This doesn't. ParserError.

```import yaml
data = yaml.load("d:{1: [18,0,0]}", yaml.Loader)
```
This doesn't either. ParserError.
